<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>inat-daily-activity-analysis - Publication Figures</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">inat-daily-activity-analysis</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./data_prep.html" rel="" target="">
 <span class="menu-text">Data Prep</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./descriptive.html" rel="" target="">
 <span class="menu-text">Descriptive</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#figure-1.-spatial-temporal-and-taxonomic-distribution-of-sscs" id="toc-figure-1.-spatial-temporal-and-taxonomic-distribution-of-sscs" class="nav-link active" data-scroll-target="#figure-1.-spatial-temporal-and-taxonomic-distribution-of-sscs">Figure 1. Spatial, temporal, and taxonomic distribution of SSCs</a></li>
  <li><a href="#figure-2.-duration-of-diel-activity-increases-with-daylength-temperature-and-wingspan" id="toc-figure-2.-duration-of-diel-activity-increases-with-daylength-temperature-and-wingspan" class="nav-link" data-scroll-target="#figure-2.-duration-of-diel-activity-increases-with-daylength-temperature-and-wingspan">Figure 2. Duration of diel activity increases with daylength, temperature, and wingspan</a></li>
  <li><a href="#figure-3.-effects-of-daylength-temperature-and-wingspan-on-onset-median-and-offset-of-diel-activity" id="toc-figure-3.-effects-of-daylength-temperature-and-wingspan-on-onset-median-and-offset-of-diel-activity" class="nav-link" data-scroll-target="#figure-3.-effects-of-daylength-temperature-and-wingspan-on-onset-median-and-offset-of-diel-activity">Figure 3. Effects of daylength, temperature, and wingspan on onset, median, and offset of diel activity</a></li>
  <li><a href="#figure-4.-activity-duration-is-the-most-phylogenetically-conserved-activity-trait" id="toc-figure-4.-activity-duration-is-the-most-phylogenetically-conserved-activity-trait" class="nav-link" data-scroll-target="#figure-4.-activity-duration-is-the-most-phylogenetically-conserved-activity-trait">Figure 4. Activity duration is the most phylogenetically conserved activity trait</a></li>
  <li><a href="#figure-5.-visualizing-sscs" id="toc-figure-5.-visualizing-sscs" class="nav-link" data-scroll-target="#figure-5.-visualizing-sscs">Figure 5. Visualizing SSCs</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Publication Figures</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Below are the steps to build the figures shown in our publication</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(colorEvoHelpers)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'dplyr::where' by 'ape::where' when loading
'colorEvoHelpers'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(splines)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rr2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'rr2' was built under R version 4.2.2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(performance)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'performance' was built under R version 4.2.1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dplyr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggpubr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggplot2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.2.2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>sscs <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"_targets/objects/sscs2"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>sscs<span class="sc">$</span>temp <span class="ot">&lt;-</span> sscs<span class="sc">$</span>value</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>sscs<span class="sc">$</span>duration <span class="ot">&lt;-</span> sscs<span class="sc">$</span>q90_value <span class="sc">-</span> sscs<span class="sc">$</span>q10_value</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>sscs<span class="sc">$</span>clade <span class="ot">&lt;-</span> sscs<span class="sc">$</span>species</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">loadFixTree</span>(<span class="at">tree_location =</span> <span class="st">"data/raw/bf_species_tree.txt"</span>,<span class="at">remove_first_split =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'stringr' was built under R version 4.2.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ape' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ape'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    rotate</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    where</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:rr2':

    binaryPGLMM</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>sscs_sp <span class="ot">&lt;-</span> sscs <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(species) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">duration=</span><span class="fu">mean</span>(duration),<span class="at">median=</span><span class="fu">mean</span>(median_value),</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">onset=</span><span class="fu">mean</span>(q10_value),<span class="at">offset=</span><span class="fu">mean</span>(q90_value),<span class="at">wingspan=</span><span class="fu">mean</span>(wingspan),<span class="at">n=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">15</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">clade=</span>species)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>obs_for_family <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"_targets/objects/obs_data"</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>species_families <span class="ot">&lt;-</span> obs_for_family <span class="sc">%&gt;%</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(family, species) <span class="sc">%&gt;%</span>  <span class="co"># Select columns of interest</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(species, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>sscs <span class="ot">&lt;-</span> <span class="fu">merge</span>(sscs,species_families,<span class="at">by=</span><span class="st">"species"</span>)</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>sscs_and_tree <span class="ot">&lt;-</span> <span class="fu">trimDfToTree</span>(sscs,tree)</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>sscs <span class="ot">&lt;-</span> sscs_and_tree[[<span class="dv">1</span>]]</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> sscs_and_tree[[<span class="dv">2</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="figure-1.-spatial-temporal-and-taxonomic-distribution-of-sscs" class="level2">
<h2 class="anchored" data-anchor-id="figure-1.-spatial-temporal-and-taxonomic-distribution-of-sscs">Figure 1. Spatial, temporal, and taxonomic distribution of SSCs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plotScatterpie</span>(sscs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'sf' was built under R version 4.2.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Linking to GEOS 3.9.3, GDAL 3.5.2, PROJ 8.2.1; sf_use_s2() is TRUE</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'rnaturalearth' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'rnaturalearthdata' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rnaturalearthdata'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:rnaturalearth':

    countries110</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggforce' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'scatterpie' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 52
[1] 463</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 3 rows containing missing values (`geom_segment()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 3 rows containing missing values (`geom_text()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="figure-2.-duration-of-diel-activity-increases-with-daylength-temperature-and-wingspan" class="level2">
<h2 class="anchored" data-anchor-id="figure-2.-duration-of-diel-activity-increases-with-daylength-temperature-and-wingspan">Figure 2. Duration of diel activity increases with daylength, temperature, and wingspan</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">getPlotLM</span>(sscs,<span class="at">model_type=</span><span class="st">"pglmm"</span>,<span class="at">bayes=</span>T,<span class="at">response=</span><span class="st">"duration"</span>,<span class="at">formula=</span><span class="st">"temp_sc + wingspan_sc + daylength_sc + (1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>,<span class="at">tree=</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'phyr' was built under R version 4.2.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'Matrix' was built under R version 4.2.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in checkDepPackageVersion(dep_pkg = "TMB"): Package version inconsistency detected.
glmmTMB was built with TMB version 1.9.0
Current TMB version is 1.9.10
Please re-install glmmTMB from source or restore original 'TMB' package (see '?reinstalling' for more information)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>as(&lt;dsCMatrix&gt;, "dgCMatrix") is deprecated since Matrix 1.5-0; do as(., "generalMatrix") instead</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The dot-dot notation (`..density..`) was deprecated in ggplot2 3.4.0.
ℹ Please use `after_stat(density)` instead.
ℹ The deprecated feature was likely used in the phyr package.
  Please report the issue at &lt;https://github.com/daijiang/phyr/issues&gt;.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'car' was built under R version 4.2.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: carData</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'carData' was built under R version 4.2.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'car'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    recode</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by Bayesian INLA

Call:duration ~ temp_sc + wingspan_sc + daylength_sc
&lt;environment: 0x000002c98e890e08&gt;

marginal logLik             DIC            WAIC 
         -14233           28136           28138 

Random effects:
                      Variance Std.Dev lower.CI upper.CI
1|cell                0.464615 0.68163 0.318061  0.63922
0 + temp_sc|cell      0.064395 0.25376 0.039707  0.12972
1|species             0.192922 0.43923 0.099114  0.35556
1|species__           0.032384 0.17995 0.010403  0.10340
0 + temp_sc|species   0.047629 0.21824 0.022947  0.09201
0 + temp_sc|species__ 0.003308 0.05752 0.000874  0.01574
residual              5.473424 2.33953 5.291186  5.61383

Fixed effects:
                Value lower.CI upper.CI
(Intercept)  7.535850 7.360267   7.7057
temp_sc      0.402378 0.235405   0.5672
wingspan_sc  0.300743 0.096354   0.5012
daylength_sc 0.265056 0.104667   0.4277

NULL</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Objects of class `communityPGLMM` are not supported.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># temperature is most reasonable </span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(rr2<span class="sc">::</span><span class="fu">R2</span>(m)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Models of class pglmm do not have a R2_resid method.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Models of class pglmm with bayes = TRUE do not have a R2_lik method.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in R2_pred(mod, mod.r, phy = phy, ...): Predictions are made with
gaussian.pred = nearest_node for models with bayes = TRUE</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  R2_pred 
0.2226848 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by Bayesian INLA

Call:duration ~ temp_sc + wingspan_sc + daylength_sc
&lt;environment: 0x000002c98e890e08&gt;

marginal logLik             DIC            WAIC 
         -14233           28136           28138 

Random effects:
                      Variance Std.Dev lower.CI upper.CI
1|cell                0.464615 0.68163 0.318061  0.63922
0 + temp_sc|cell      0.064395 0.25376 0.039707  0.12972
1|species             0.192922 0.43923 0.099114  0.35556
1|species__           0.032384 0.17995 0.010403  0.10340
0 + temp_sc|species   0.047629 0.21824 0.022947  0.09201
0 + temp_sc|species__ 0.003308 0.05752 0.000874  0.01574
residual              5.473424 2.33953 5.291186  5.61383

Fixed effects:
                Value lower.CI upper.CI
(Intercept)  7.535850 7.360267   7.7057
temp_sc      0.402378 0.235405   0.5672
wingspan_sc  0.300743 0.096354   0.5012
daylength_sc 0.265056 0.104667   0.4277</code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># effect of temperature</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname =</span> <span class="st">"duration"</span>, <span class="at">pred_colname=</span><span class="st">"temp_sc"</span>, <span class="at">other_pred_colnames =</span> <span class="fu">c</span>(<span class="st">"daylength_sc"</span>,<span class="st">"wingspan_sc"</span>), </span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames =</span> <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyverse' was built under R version 4.2.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2
──</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ tibble  3.2.1     ✔ purrr   1.0.1
✔ tidyr   1.3.0     ✔ forcats 0.5.2
✔ readr   2.1.2     </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tibble' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyr' was built under R version 4.2.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'readr' was built under R version 4.2.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'purrr' was built under R version 4.2.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'forcats' was built under R version 4.2.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ tidyr::expand() masks Matrix::expand()
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
✖ tidyr::pack()   masks Matrix::pack()
✖ car::recode()   masks dplyr::recode()
✖ purrr::some()   masks car::some()
✖ tidyr::unpack() masks Matrix::unpack()
✖ ape::where()    masks dplyr::where()</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> p1 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Temperature"</span>, <span class="at">y =</span> <span class="st">"Daily Activity Duration (Hours)"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> temp_sc, <span class="at">y =</span> duration),<span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="co"># effect of daylength</span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname =</span> <span class="st">"duration"</span>, <span class="at">pred_colname=</span><span class="st">"daylength_sc"</span>, <span class="at">other_pred_colnames =</span> <span class="fu">c</span>(<span class="st">"temp_sc"</span>,<span class="st">"wingspan_sc"</span>), </span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames =</span> <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Daylength"</span>, <span class="at">y =</span> <span class="st">"Duration"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> daylength_sc, <span class="at">y =</span> duration),<span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a><span class="co"># effect of wingspan</span></span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname =</span> <span class="st">"duration"</span>, <span class="at">pred_colname=</span><span class="st">"wingspan_sc"</span>, <span class="at">other_pred_colnames =</span> <span class="fu">c</span>(<span class="st">"daylength_sc"</span>,<span class="st">"temp_sc"</span>), </span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames =</span> <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> p3 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Wingspan"</span>, <span class="at">y =</span> <span class="st">"Duration"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> wingspan_sc, <span class="at">y =</span> duration),<span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> p3 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(p1, p2, p3, <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">nrow =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="figure-3.-effects-of-daylength-temperature-and-wingspan-on-onset-median-and-offset-of-diel-activity" class="level2">
<h2 class="anchored" data-anchor-id="figure-3.-effects-of-daylength-temperature-and-wingspan-on-onset-median-and-offset-of-diel-activity">Figure 3. Effects of daylength, temperature, and wingspan on onset, median, and offset of diel activity</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>rmPlotAxis <span class="ot">&lt;-</span> <span class="cf">function</span>(p, <span class="at">rm_x=</span><span class="cn">FALSE</span>, <span class="at">rm_y=</span><span class="cn">FALSE</span>){</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(rm_y){</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(rm_x){</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p)</span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a><span class="do">## onset</span></span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">getPlotLM</span>(sscs,<span class="at">model_type=</span><span class="st">"pglmm"</span>,<span class="at">bayes=</span>T,<span class="at">response=</span><span class="st">"q10_value"</span>,<span class="at">formula=</span><span class="st">"temp_sc + wingspan_sc + daylength_sc + (1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>,<span class="at">tree=</span>tree)<span class="co">#, data = sscs, cov_ranef = list(species=tree), bayes=TRUE)# print(vif(m))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by Bayesian INLA

Call:q10_value ~ temp_sc + wingspan_sc + daylength_sc
&lt;environment: 0x000002c9927c8f58&gt;

marginal logLik             DIC            WAIC 
         -12564           24762           24757 

Random effects:
                       Variance  Std.Dev  lower.CI  upper.CI
1|cell                1.819e+00 1.348645 1.599e+00 2.0766409
0 + temp_sc|cell      2.362e-02 0.153696 1.256e-02 0.0575767
1|species             3.546e-02 0.188305 1.578e-02 0.0906553
1|species__           9.341e-03 0.096651 3.719e-03 0.0275801
0 + temp_sc|species   5.618e-03 0.074954 2.673e-03 0.0129578
0 + temp_sc|species__ 5.105e-05 0.007145 1.936e-05 0.0001502
residual              3.182e+00 1.783821 3.053e+00 3.3237698

Fixed effects:
                 Value  lower.CI upper.CI
(Intercept)  10.248203 10.095841  10.4038
temp_sc      -0.078420 -0.244862   0.0926
wingspan_sc  -0.032679 -0.128116   0.0671
daylength_sc -0.405445 -0.592831  -0.2223

NULL</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Objects of class `communityPGLMM` are not supported.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(rr2<span class="sc">::</span><span class="fu">R2</span>(m))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Models of class pglmm do not have a R2_resid method.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Models of class pglmm with bayes = TRUE do not have a R2_lik method.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in R2_pred(mod, mod.r, phy = phy, ...): Predictions are made with
gaussian.pred = nearest_node for models with bayes = TRUE</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  R2_pred 
0.3796571 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by Bayesian INLA

Call:q10_value ~ temp_sc + wingspan_sc + daylength_sc
&lt;environment: 0x000002c9927c8f58&gt;

marginal logLik             DIC            WAIC 
         -12564           24762           24757 

Random effects:
                       Variance  Std.Dev  lower.CI  upper.CI
1|cell                1.819e+00 1.348645 1.599e+00 2.0766409
0 + temp_sc|cell      2.362e-02 0.153696 1.256e-02 0.0575767
1|species             3.546e-02 0.188305 1.578e-02 0.0906553
1|species__           9.341e-03 0.096651 3.719e-03 0.0275801
0 + temp_sc|species   5.618e-03 0.074954 2.673e-03 0.0129578
0 + temp_sc|species__ 5.105e-05 0.007145 1.936e-05 0.0001502
residual              3.182e+00 1.783821 3.053e+00 3.3237698

Fixed effects:
                 Value  lower.CI upper.CI
(Intercept)  10.248203 10.095841  10.4038
temp_sc      -0.078420 -0.244862   0.0926
wingspan_sc  -0.032679 -0.128116   0.0671
daylength_sc -0.405445 -0.592831  -0.2223</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># effect of temp</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m,sscs,tree,<span class="at">response_colname =</span> <span class="st">"q10_value"</span>,<span class="at">pred_colname =</span> <span class="st">"temp_sc"</span>,<span class="at">other_pred_colnames =</span> <span class="fu">c</span>(<span class="st">"wingspan_sc"</span>,<span class="st">"daylength_sc"</span>), </span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames =</span> <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> p1 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Temperature"</span>, <span class="at">y =</span> <span class="st">"Onset"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> temp_sc, <span class="at">y =</span> q10_value),<span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a><span class="co"># effect of daylength</span></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m,sscs,tree,<span class="at">response_colname =</span> <span class="st">"q10_value"</span>,<span class="at">pred_colname =</span> <span class="st">"daylength_sc"</span>,<span class="at">other_pred_colnames =</span> <span class="fu">c</span>(<span class="st">"wingspan_sc"</span>,<span class="st">"temp_sc"</span>), </span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames =</span> <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Daylength"</span>, <span class="at">y =</span> <span class="st">"Onset"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> daylength_sc, <span class="at">y =</span> q10_value),<span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">rmPlotAxis</span>(p2,<span class="at">rm_y=</span><span class="cn">TRUE</span>)</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a><span class="co"># effect of wingspan </span></span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m,sscs,tree,<span class="at">response_colname =</span> <span class="st">"q10_value"</span>,<span class="at">pred_colname =</span> <span class="st">"wingspan_sc"</span>,<span class="at">other_pred_colnames =</span> <span class="fu">c</span>(<span class="st">"temp_sc"</span>,<span class="st">"daylength_sc"</span>),</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames =</span> <span class="fu">c</span>(<span class="st">"species"</span>,<span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> p3 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Wingspan"</span>, <span class="at">y =</span> <span class="st">"Onset"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> wingspan_sc, <span class="at">y =</span> q10_value),<span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> p3 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">rmPlotAxis</span>(p3,<span class="at">rm_y=</span><span class="cn">TRUE</span>)</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a><span class="do">## median</span></span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">getPlotLM</span>(sscs, <span class="at">model_type=</span><span class="st">"pglmm"</span>, <span class="at">bayes=</span>T, <span class="at">response=</span><span class="st">"median_value"</span>, <span class="at">formula=</span><span class="st">"temp_sc + wingspan_sc + daylength_sc + (1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>, <span class="at">tree=</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by Bayesian INLA

Call:median_value ~ temp_sc + wingspan_sc + daylength_sc
&lt;environment: 0x000002c98ed61908&gt;

marginal logLik             DIC            WAIC 
         -12487           24373           24398 

Random effects:
                      Variance Std.Dev lower.CI upper.CI
1|cell                4.715998 2.17163 3.493442  6.91940
0 + temp_sc|cell      0.093812 0.30629 0.062777  0.15863
1|species             0.130240 0.36089 0.077700  0.26211
1|species__           0.021965 0.14821 0.011787  0.04545
0 + temp_sc|species   0.075613 0.27498 0.047206  0.11799
0 + temp_sc|species__ 0.006985 0.08357 0.003237  0.01650
residual              2.941175 1.71499 2.849423  3.03823

Fixed effects:
                 Value  lower.CI upper.CI
(Intercept)  13.801627 13.617343  13.9862
temp_sc       0.124807 -0.088472   0.3405
wingspan_sc   0.072232 -0.079528   0.2274
daylength_sc -0.307055 -0.524460  -0.0907

NULL</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Objects of class `communityPGLMM` are not supported.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-4-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(rr2<span class="sc">::</span><span class="fu">R2</span>(m))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Models of class pglmm do not have a R2_resid method.
Models of class pglmm with bayes = TRUE do not have a R2_lik method.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in R2_pred(mod, mod.r, phy = phy, ...): Predictions are made with
gaussian.pred = nearest_node for models with bayes = TRUE</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  R2_pred 
0.6250751 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by Bayesian INLA

Call:median_value ~ temp_sc + wingspan_sc + daylength_sc
&lt;environment: 0x000002c98ed61908&gt;

marginal logLik             DIC            WAIC 
         -12487           24373           24398 

Random effects:
                      Variance Std.Dev lower.CI upper.CI
1|cell                4.715998 2.17163 3.493442  6.91940
0 + temp_sc|cell      0.093812 0.30629 0.062777  0.15863
1|species             0.130240 0.36089 0.077700  0.26211
1|species__           0.021965 0.14821 0.011787  0.04545
0 + temp_sc|species   0.075613 0.27498 0.047206  0.11799
0 + temp_sc|species__ 0.006985 0.08357 0.003237  0.01650
residual              2.941175 1.71499 2.849423  3.03823

Fixed effects:
                 Value  lower.CI upper.CI
(Intercept)  13.801627 13.617343  13.9862
temp_sc       0.124807 -0.088472   0.3405
wingspan_sc   0.072232 -0.079528   0.2274
daylength_sc -0.307055 -0.524460  -0.0907</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Effect of temperature on median_value</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname=</span><span class="st">"median_value"</span>, <span class="at">pred_colname=</span><span class="st">"temp_sc"</span>, <span class="at">other_pred_colnames=</span><span class="fu">c</span>(<span class="st">"wingspan_sc"</span>, <span class="st">"daylength_sc"</span>), </span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames=</span><span class="fu">c</span>(<span class="st">"species"</span>, <span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> p4 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Temperature"</span>, <span class="at">y =</span> <span class="st">"Peak"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> temp_sc, <span class="at">y =</span> median_value), <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">rmPlotAxis</span>(p4,<span class="at">rm_x=</span><span class="cn">TRUE</span>)</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Effect of daylength on median_value</span></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname=</span><span class="st">"median_value"</span>, <span class="at">pred_colname=</span><span class="st">"daylength_sc"</span>, <span class="at">other_pred_colnames=</span><span class="fu">c</span>(<span class="st">"wingspan_sc"</span>, <span class="st">"temp_sc"</span>), </span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames=</span><span class="fu">c</span>(<span class="st">"species"</span>, <span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> p5 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Daylength"</span>, <span class="at">y =</span> <span class="st">"Peak"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> daylength_sc, <span class="at">y =</span> median_value), <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">rmPlotAxis</span>(p5,<span class="at">rm_x=</span><span class="cn">TRUE</span>,<span class="at">rm_y=</span><span class="cn">TRUE</span>)</span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Effect of wingspan on median_value</span></span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname=</span><span class="st">"median_value"</span>, <span class="at">pred_colname=</span><span class="st">"wingspan_sc"</span>, <span class="at">other_pred_colnames=</span><span class="fu">c</span>(<span class="st">"temp_sc"</span>, <span class="st">"daylength_sc"</span>),</span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames=</span><span class="fu">c</span>(<span class="st">"species"</span>, <span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> p6 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Wingspan"</span>, <span class="at">y =</span> <span class="st">"Peak"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> wingspan_sc, <span class="at">y =</span> median_value), <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">rmPlotAxis</span>(p6,<span class="at">rm_x=</span><span class="cn">TRUE</span>,<span class="at">rm_y=</span><span class="cn">TRUE</span>)</span>
<span id="cb92-16"><a href="#cb92-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-17"><a href="#cb92-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-18"><a href="#cb92-18" aria-hidden="true" tabindex="-1"></a><span class="do">## offset</span></span>
<span id="cb92-19"><a href="#cb92-19" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">getPlotLM</span>(sscs, <span class="at">model_type=</span><span class="st">"pglmm"</span>, <span class="at">bayes=</span>T, <span class="at">response=</span><span class="st">"q90_value"</span>, <span class="at">formula=</span><span class="st">"temp_sc + wingspan_sc + daylength_sc + (1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>, <span class="at">tree=</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by Bayesian INLA

Call:q90_value ~ temp_sc + wingspan_sc + daylength_sc
&lt;environment: 0x000002ca5f8b6910&gt;

marginal logLik             DIC            WAIC 
         -12661           24812           24846 

Random effects:
                      Variance Std.Dev lower.CI upper.CI
1|cell                 1.70298  1.3050 1.218304  2.46981
0 + temp_sc|cell       0.05753  0.2399 0.038733  0.09411
1|species              0.21038  0.4587 0.108788  0.43022
1|species__            0.02182  0.1477 0.013094  0.03847
0 + temp_sc|species    0.08991  0.2998 0.048254  0.21360
0 + temp_sc|species__  0.01746  0.1321 0.008395  0.03766
residual               3.17802  1.7827 3.027280  3.42715

Fixed effects:
                 Value  lower.CI upper.CI
(Intercept)  17.705725 17.536683  17.8738
temp_sc       0.205329  0.015413   0.3951
wingspan_sc   0.283905  0.130835   0.4384
daylength_sc  0.028000 -0.158809   0.2151

NULL</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Objects of class `communityPGLMM` are not supported.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-4-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(rr2<span class="sc">::</span><span class="fu">R2</span>(m))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Models of class pglmm do not have a R2_resid method.
Models of class pglmm with bayes = TRUE do not have a R2_lik method.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in R2_pred(mod, mod.r, phy = phy, ...): Predictions are made with
gaussian.pred = nearest_node for models with bayes = TRUE</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  R2_pred 
0.4621541 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by Bayesian INLA

Call:q90_value ~ temp_sc + wingspan_sc + daylength_sc
&lt;environment: 0x000002ca5f8b6910&gt;

marginal logLik             DIC            WAIC 
         -12661           24812           24846 

Random effects:
                      Variance Std.Dev lower.CI upper.CI
1|cell                 1.70298  1.3050 1.218304  2.46981
0 + temp_sc|cell       0.05753  0.2399 0.038733  0.09411
1|species              0.21038  0.4587 0.108788  0.43022
1|species__            0.02182  0.1477 0.013094  0.03847
0 + temp_sc|species    0.08991  0.2998 0.048254  0.21360
0 + temp_sc|species__  0.01746  0.1321 0.008395  0.03766
residual               3.17802  1.7827 3.027280  3.42715

Fixed effects:
                 Value  lower.CI upper.CI
(Intercept)  17.705725 17.536683  17.8738
temp_sc       0.205329  0.015413   0.3951
wingspan_sc   0.283905  0.130835   0.4384
daylength_sc  0.028000 -0.158809   0.2151</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Effect of temperature on q90_value</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname=</span><span class="st">"q90_value"</span>, <span class="at">pred_colname=</span><span class="st">"temp_sc"</span>, <span class="at">other_pred_colnames=</span><span class="fu">c</span>(<span class="st">"wingspan_sc"</span>, <span class="st">"daylength_sc"</span>), </span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames=</span><span class="fu">c</span>(<span class="st">"species"</span>, <span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) +s (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> p7 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Temperature"</span>, <span class="at">y =</span> <span class="st">"Offset"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> temp_sc, <span class="at">y =</span> q90_value), <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> <span class="fu">rmPlotAxis</span>(p7,<span class="at">rm_x=</span><span class="cn">TRUE</span>)</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Effect of daylength on q90_value</span></span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>p8 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname=</span><span class="st">"q90_value"</span>, <span class="at">pred_colname=</span><span class="st">"daylength_sc"</span>, <span class="at">other_pred_colnames=</span><span class="fu">c</span>(<span class="st">"wingspan_sc"</span>, <span class="st">"temp_sc"</span>), </span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames=</span><span class="fu">c</span>(<span class="st">"species"</span>, <span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a>p8 <span class="ot">&lt;-</span> p8 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Daylength"</span>, <span class="at">y =</span> <span class="st">"Offset"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> daylength_sc, <span class="at">y =</span> q90_value), <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a>p8 <span class="ot">&lt;-</span> <span class="fu">rmPlotAxis</span>(p8,<span class="at">rm_y=</span><span class="cn">TRUE</span>,<span class="at">rm_x=</span><span class="cn">TRUE</span>)</span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Effect of wingspan on q90_value</span></span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a>p9 <span class="ot">&lt;-</span> <span class="fu">plotPglmmEffect</span>(m, sscs, tree, <span class="at">response_colname=</span><span class="st">"q90_value"</span>, <span class="at">pred_colname=</span><span class="st">"wingspan_sc"</span>, <span class="at">other_pred_colnames=</span><span class="fu">c</span>(<span class="st">"temp_sc"</span>, <span class="st">"daylength_sc"</span>),</span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">random_effect_colnames=</span><span class="fu">c</span>(<span class="st">"species"</span>, <span class="st">"cell"</span>), <span class="at">random_effects_formula=</span><span class="st">"(1|cell) + (0+temp_sc|cell) + (1|species__) + (0+temp_sc|species__)"</span>)</span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a>p9 <span class="ot">&lt;-</span> p9 <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Wingspan"</span>, <span class="at">y =</span> <span class="st">"Offset"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> sscs, <span class="fu">aes</span>(<span class="at">x =</span> wingspan_sc, <span class="at">y =</span> q90_value), <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb102-15"><a href="#cb102-15" aria-hidden="true" tabindex="-1"></a>p9 <span class="ot">&lt;-</span> p9 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb102-16"><a href="#cb102-16" aria-hidden="true" tabindex="-1"></a>p9 <span class="ot">&lt;-</span> <span class="fu">rmPlotAxis</span>(p9,<span class="at">rm_y=</span><span class="cn">TRUE</span>,<span class="at">rm_x=</span><span class="cn">TRUE</span>)</span>
<span id="cb102-17"><a href="#cb102-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-18"><a href="#cb102-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(p7,p8,p9,p4,p5,p6,p1,p2,p3, <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-4-4.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="figure-4.-activity-duration-is-the-most-phylogenetically-conserved-activity-trait" class="level2">
<h2 class="anchored" data-anchor-id="figure-4.-activity-duration-is-the-most-phylogenetically-conserved-activity-trait">Figure 4. Activity duration is the most phylogenetically conserved activity trait</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>sscs_sp <span class="ot">&lt;-</span> sscs <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(species) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">duration=</span><span class="fu">mean</span>(duration),<span class="at">median=</span><span class="fu">mean</span>(median_value),</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">onset=</span><span class="fu">mean</span>(q10_value),<span class="at">offset=</span><span class="fu">mean</span>(q90_value),<span class="at">wingspan=</span><span class="fu">mean</span>(wingspan),<span class="at">n=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">15</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">clade=</span>species)</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a><span class="fu">assessTraitsOnTree</span>(sscs_sp, <span class="at">trait_colname =</span> <span class="st">"duration"</span>,<span class="at">tree=</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: maps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'maps'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    map</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'phytools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ape':

    drop.tip.multiPhylo</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'caper' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MASS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MASS'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: mvtnorm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>legend scale cannot be longer than total tree length; resetting</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal K : 0.141633 
P-value (based on 1000 randomizations) : 0.74 </code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal lambda : 0.218009 
logL(lambda) : -90.2409 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assessTraitsOnTree</span>(sscs_sp, <span class="at">trait_colname =</span> <span class="st">"onset"</span>,<span class="at">tree=</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>legend scale cannot be longer than total tree length; resetting</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal K : 0.126228 
P-value (based on 1000 randomizations) : 0.895 </code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal lambda : 6.6107e-05 
logL(lambda) : -97.0204 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assessTraitsOnTree</span>(sscs_sp, <span class="at">trait_colname =</span> <span class="st">"median"</span>,<span class="at">tree=</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>legend scale cannot be longer than total tree length; resetting</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal K : 0.109857 
P-value (based on 1000 randomizations) : 0.973 </code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal lambda : 6.6107e-05 
logL(lambda) : -137.006 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assessTraitsOnTree</span>(sscs_sp, <span class="at">trait_colname =</span> <span class="st">"offset"</span>,<span class="at">tree=</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>legend scale cannot be longer than total tree length; resetting</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-11.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal K : 0.112471 
P-value (based on 1000 randomizations) : 0.949 </code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-5-12.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal lambda : 6.6107e-05 
logL(lambda) : -121.087 </code></pre>
</div>
</div>
<p>Wingspan and duration are phylogenetically correlated</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(sscs_sp<span class="sc">$</span>duration,sscs_sp<span class="sc">$</span>wingspan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Pearson's product-moment correlation

data:  sscs_sp$duration and sscs_sp$wingspan
t = 3.3574, df = 68, p-value = 0.001292
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.1559269 0.5622415
sample estimates:
     cor 
0.377085 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assessTraitsOnTree</span>(sscs_sp, <span class="at">trait_colname =</span> <span class="st">"duration"</span>,<span class="at">trait_colname2=</span><span class="st">"wingspan"</span>,<span class="at">tree=</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>legend scale cannot be longer than total tree length; resetting</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal K : 0.141633 
P-value (based on 1000 randomizations) : 0.743 </code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal lambda : 0.218009 
logL(lambda) : -90.2409 </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>legend scale cannot be longer than total tree length; resetting</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-6-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-6-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-6-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal K : 1.46274 
P-value (based on 1000 randomizations) : 0.001 </code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-6-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Phylogenetic signal lambda : 0.924629 
logL(lambda) : -116.561 </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'nlme'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:lme4':

    lmList</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    collapse</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in Initialize.corPhyl(X[[i]], ...): No covariate specified, species
will be taken as ordered in the data frame. To avoid this message, specify a
covariate containing the species names with the 'form' argument.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Denom. DF: 68 
            numDF   F-value p-value
(Intercept)     1 184.30424  &lt;.0001
wingspan        1  19.93224  &lt;.0001</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Converting into numeric values currently not possible for variables of
  class 'NULL'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in mean.default(resp, na.rm = TRUE): argument is not numeric or
logical: returning NA</code></pre>
</div>
<div class="cell-output-display">
<p><img src="publication_figures_files/figure-html/unnamed-chunk-6-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  R2: NaN</code></pre>
</div>
</div>
</section>
<section id="figure-5.-visualizing-sscs" class="level2">
<h2 class="anchored" data-anchor-id="figure-5.-visualizing-sscs">Figure 5. Visualizing SSCs</h2>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>